src/SonarScanner.MSBuild.Tasks/WriteProjectConfigFile.cs

```diff
@@ -0,0 +1,82 @@
+﻿/*
+ * SonarScanner for MSBuild
+ * Copyright (C) 2016-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.IO;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using SonarScanner.MSBuild.Common;
+
+namespace SonarScanner.MSBuild.Tasks
+{
+    /// <summary>
+    /// MSBuild task to write a SonarProjectConfig file to disk in XML format
+    /// </summary>
+    /// <remarks>SonarProjectConfig.xml file is used to pass information from Scanner to the Analyzers.</remarks>
+    public class WriteProjectConfigFile : Task
+    {
+        #region Input properties
+
+        /// <summary>
+        /// Project specific config directory where SonarProjectConfig.xml will be saved.
+        /// </summary>
+        [Required]
+        public string ConfigDir { get; set; }
+
+        public string AnalysisConfigPath { get; set; }
```

Comment:
aren't all `Required`? Why is only `ConfigDir` required?

I see you are testing the missing parameters in `Execute_IsTest_FalseReturnsTypeProduct` and other tests, but I'm not sure if this Task should allow for this usage?

Should this Task just disallow usage without all provided parameters?

Reply:
(and test this scenario)

Reply:
Only `ConfigDir` is required for the task to do it's job. If anything of the rest is empty, it will be written as empty and analyzer can live with such state.

_Side note:_ There's test `Execute_AllPropertiesAreSet` that's checking if all is set, but this one ensures that all arguments are properly propagated to the XML.